packages/features/users/components/UserTable/BulkActions/EventTypesList.tsx

```diff
@@ -0,0 +1,262 @@
+
+                    const ids = events.map((event) => event.id);
+                    const areAllUsersHostForTeam = selectedUsers.every((user) =>
+                      events.every((event) => event.hosts.some((host) => host.userId === user.id))
+                    );
+                    const isSelected = ids.every(
+                      (id) =>
+                        selectedEvents.has(id) || (areAllUsersHostForTeam && !removeHostFromEvents.has(id))
+                    );
+                    return (
+                      <>
+                        <ListItem
+                          isTeam
+                          onSelect={() => {
+                            if (!isSelected) {
+                              // Add current team and its event
+                              addValue(selectedTeams, setSelectedTeams, [teamId]);
+                              addValue(selectedEvents, setSelectedEvents, ids);
+                              setRemoveHostFromEvents(new Set());
+                            } else {
+                              const eventIdsWhereAllUsersAreHosts = events
+                                .filter((event) =>
+                                  selectedUsers.every((user) =>
+                                    event.hosts.some((host) => host.userId === user.id)
+                                  )
+                                )
+                                .map((event) => event.id);
+
+                              addValue(
+                                removeHostFromEvents,
+                                setRemoveHostFromEvents,
+                                eventIdsWhereAllUsersAreHosts
+                              );
+                              // Remove selected team and its event
+                              removeValue(selectedEvents, setSelectedEvents, ids);
+                              removeValue(selectedTeams, setSelectedTeams, [teamId]);
+                            }
+                          }}
+                          isSelected={isSelected}
+                          text={team.profile.name || ""}
+                          key={team.profile.name}
+                        />
+                        {events.map((event) => {
+                          const hosts = event.hosts;
+                          const areAllUsersHostForEventType = selectedUsers.every((user) =>
+                            hosts.some((host) => host.userId === user.id)
+                          );
+                          const isSelected =
+                            (selectedEvents.has(event.id) || areAllUsersHostForEventType) &&
+                            !removeHostFromEvents.has(event.id);
+                          return (
+                            <ListItem
+                              isTeam={false}
+                              onSelect={() => {
+                                if (!isSelected) {
+                                  if (areAllUsersHostForEventType) {
```

Comment:
If all selected users are host and it is still unchecked, then it was because it was added to removeEvents, so remove it from there